benchexec/tablegenerator/react-table/src/utils/utils.js

```diff
@@ -203,68 +203,97 @@ const EXTENDED_DISCRETE_COLOR_RANGE = [
-    const removedItems = Object.entries(params).filter(
-      ([_, value]) => value === undefined || value === null,
-    );
-    removedItems.forEach(([key]) => {
-      delete additionalParams[key];
-      delete transformedParams[key];
-    });
-  }
+export const constructQueryString = (params) => {
+  return Object.keys(params)
+    .filter((key) => params[key] !== undefined && params[key] !== null)
+    .map((key) => `${key}=${params[key]}`)
+    .join("&");
+};
+
+/**
+ * Constructs a URL hash from the provided parameters
+ *
+ * @param {string} url - The URL to be processed
+ * @param {Object} params - The parameters to be included in the hash
+ * @param {boolean} [keepOthers=false] - Whether to keep existing parameters in the URL hash or not
+ * @returns {string} - The constructed URL hash
+ */
+export const constructHashURL = (url, params = {}, keepOthers = false) => {
+  const additionalParams = keepOthers ? getHashSearch(url) : {};
   const mergedParams = { ...additionalParams, ...params };
-  const optionTemplate = { returnString: false, baseUrl: null };
-  const { returnString, baseUrl, history } = { ...optionTemplate, ...options };
-  const url = (baseUrl || document.location.href).split("?")[0];
-  const pairs = Object.keys(mergedParams).map(
-    (key) => `${key}=${mergedParams[key]}`,
-  );
-  const searchString = `?${pairs.join("&")}`;
-  const hrefString = encodeURI(`${url}${searchString}`);
-  if (returnString) {
-    return hrefString;
+
+  const queryString = constructQueryString(mergedParams);
+  const baseURL = url.split("?")[0];
+
+  return queryString.length > 0 ? `${baseURL}?${queryString}` : baseURL;
+};
+
+/**
+ * Sets or updates the search parameters in the URL hash of the current page. All the existing search parameters will not be disturbed. Also accepts a history object to update the URL hash without reloading the page.
+ *
+ * @param {Object} params - The parameters to be set or updated in the URL hash
+ * @param {Object} [history=null] - The history object to use for updating the URL hash
+ * @returns {void}
+ */
+const setParam = (params = {}, history = null) => {
+  const keepOthers = true;
+  const newUrl = constructHashURL(document.location.href, params, keepOthers);
+
+  if (history && history.push) {
+    history.push(newUrl);
   }
-  if (history) {
-    history.push(searchString);
-    return;
+  document.location.href = newUrl;
+};
+
+/**
+ * Removes the specified search parameter from the URL hash of the current page. Also accepts a history object to update the URL hash without reloading the page.
+ * @param {string} param - The name of the parameter to be removed
+ * @param {Object} [history=null] - The history object to use for updating the URL hash
+ */
+export const removeParam = (param, history = null) => {
+  const params = getHashSearch();
+  delete params[param];
+
+  const keepOthers = false;
+  const newUrl = constructHashURL(document.location.href, params, keepOthers);
```

Comment:
Isn't this the same as `constructHashURL(document.location.href, {param: undefined}, true)`? `constructHashURL ` first sets the parameter to `undefined` and then `constructQueryString` removes it from the result?

If I am not mistaken, we could remove `removeParam` and document that `setParam` can be used to delete a parameter by setting it to `undefined`. Or alternatively we can replace the content of `removeParam` with `setParam({param: undefined}, history)`, but I don't think it is necessary to keep a separate method. Also we can remove the `keepOthers` parameter from `constructHashURL`.

Reply:
Ahh, nice! This trick to remove parameters is super neat! I'll update the implementation and document this behavior in the JSDoc comments. 

Though I don't think we should deprecate the `keepOthers` parameter from the `constructHashURL` function, as it may be needed in the future? We already have a comprehensive test suite for it, and the same is only being utilized once in the codebase, so it might be good to keep support for the same. How do you feel about it?

Reply:
Yes, the tests are good, but I don't really see why we would possibly need it in the future. In the end, it has turned out that all existing users of this feature don't actually need it. In general I think we can prefer less complexity over stuff that is only hypothetically useful in the future. It is also not complex to re-add if we ever need it.

Reply:
Okay sir!